pyteal/ast/subroutine.py

```diff
@@ -304,6 +306,7 @@ def _validate_annotation(
         )
 
     def get_declaration(self) -> "SubroutineDeclaration":
+        """MARK AS DEPRECATED, FOR WE ARE NOW GET DECLARATION BY COMPILE OPTIONS"""
```

Comment:
@ahangsu Checking if it's been considered:  Since the docstring mentions deprecation, have you also considered adding a `warnings.warn(..., ..., DeprecationWarning`) to formally announce deprecation? 

Presumably, adding a formal deprecation paves the way for us to remove these methods in a future release.

Reply:
That makes sense, just like py-sdk.